src/components/common/Map.js

```diff
@@ -0,0 +1,129 @@
+import React, { PropTypes } from 'react';
+import { GoogleMapLoader, GoogleMap, Marker } from 'react-google-maps';
+import { getMyPosition } from '../../utils/MapHelper';
+
+class Map extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.onMapClick = this.onMapClick.bind(this);
+    this.success = this.success.bind(this);
+    this.error = this.error.bind(this);
+    this.getOpts = this.getOpts.bind(this);
+    this.handleMapMounted = this.handleMapMounted.bind(this);
+    this.handleCenterChanged = this.handleCenterChanged.bind(this);
+
+    const myPosCoords = JSON.parse(localStorage.getItem('myPositionCoords'));
+    if (!myPosCoords) {
+      getMyPosition(this.success, this.error);
+      this.state = {
+        locationCenter: this.props.center
+      };
+    }else {
+      this.state = {
+        locationCenter: { position: myPosCoords }
+      };
+    }
+  }
+
+  handleMapMounted(map) {
+    this._map = map;
+  }
+
+  handleCenterChanged() {
+    const mapMovement = this._map.getCenter();
+    let nextCenter = targetSelected || {
+      lat: parseFloat(mapMovement.lat().toFixed(6)),
+      lng: parseFloat(mapMovement.lng().toFixed(6))
+    };
+
+    let newState = Object.assign({}, this.state);
+    newState.locationCenter.lat = nextCenter.lat;
+    newState.locationCenter.lng = nextCenter.lng;
+    if (newState.locationCenter.lat === this.state.locationCenter.lat &&
+      newState.locationCenter.lng === this.state.locationCenter.lng &&
+      targetSelected === null) {
+      // Notice: Check nextCenter equality here,
+      // or it will fire center_changed event infinitely
+      return;
+    }
+    this.setState(newState);
+  }
+
+  getOpts(venue) {
```

Comment:
You can write this function in much less lines by declaring some objects in one line. For example:

`origin: { x: 0, y: 0 },`